frontend/src/metabase/query_builder/components/chart-type-selector/use-chart-type-visualizations.ts

```diff
@@ -0,0 +1,94 @@
+import { useCallback, useMemo } from "react";
+import _ from "underscore";
+
+import visualizations from "metabase/visualizations";
+import { sanatizeResultData } from "metabase/visualizations/shared/utils/data";
+import type Question from "metabase-lib/v1/Question";
+import {
+  type CardDisplayType,
+  type Dataset,
+  type VisualizationDisplay,
+  isCardDisplayType,
+} from "metabase-types/api";
+
+import { DEFAULT_ORDER } from "./viz-order";
+
+export type UseChartTypeVisualizationsProps = {
+  question: Question;
+  onUpdateQuestion: (question: Question) => void;
+};
+
+export const useChartTypeVisualizations = ({
+  question,
+  onUpdateQuestion,
+  result,
+}: UseChartTypeVisualizationsProps & GetSensibleVisualizationsProps) => {
+  const selectedVisualization = question.display();
+
+  const updateQuestionVisualization = useCallback(
+    (display: CardDisplayType) => {
+      let newQuestion = question.setDisplay(display).lockDisplay(); // prevent viz auto-selection
+      const visualization = visualizations.get(display);
+      if (visualization?.onDisplayUpdate) {
+        const updatedSettings = visualization.onDisplayUpdate(
+          newQuestion.settings(),
+        );
+        newQuestion = newQuestion.setSettings(updatedSettings);
+        onUpdateQuestion(question);
+      }
```

Comment:
nit: we could get rid of the temp variable by moving `newQuestion` below

```suggestion
      const visualization = visualizations.get(display);
      
      if (visualization?.onDisplayUpdate) {
        // prevent viz auto-selection
        const newQuestion = question.setDisplay(display).lockDisplay();
        const updatedSettings = visualization.onDisplayUpdate(
          newQuestion.settings(),
        );
        
        onUpdateQuestion(newQuestion.setSettings(updatedSettings));
      }
```